### ISSUE TITLE
LTS version

### ISSUE THREAD

**Issue by farhanianz** [EFTEC/BladeOne#114]
I appreciate your hard work for the repo. It's a nice piece of real standalone library. I was thinking about a more stable & long term support version of it, since it may help on certain situation. Currently it has only minimal known bugs which could be taken care of if coded carefully for now at least. What your opinion about releasing a LTS version of BladeOne? If positive, when we can expect that to be released? Or what are the reasons not having a LTS version if you think it is not necessary? 
Thanks.

**Comment by jorgecc** [comment]
Hi there:

* First and foremost, **the support** is limited to my spare time :-)    So that is the support of this library, LTS or not.

As always, If you or anybody want to collaborate, they are welcome.

* Second, the goal of this library is to support the templates of Blade without Laravel. Why? long story, short, Laravel sucks.   

✔It is fulfilled.

* Third, this library is always aimed to work in tandem with the programmer - designers.  So it is the reason why I liked the notations of Blade (or Razor in c#), it's simple and it works.

✔It is fulfilled.

* Fourth is the speed and this library could be pushed a bit more.  I am working on the **cache library** and I am not decided if I want to integrate into the "core" or separate into a different library, as I did with BladeOneHtml.

📅 It is pending.

* Fifth, Functionality: Laravel's Blade added a new feature called "X-Components".   BladeOne specifically does not support it because the implementation of components is slow and it's a different way to work (it doesn't fit in the ideology of Blade).

It is components:

```html
<x-alert type="error" :message="$message" class="mt-4"/>
```

BladeOne has several ways to create new reusable components so I don't think to implement components, for example, if we create a new tag called xalert:

```html
@xalert("error",$message,"mt-4")
```

While it does the same but what if we only indicate the class attribute?.  In Blade's component will look like:

```html
<x-alert class="mt-4"/>
```

while in BladeOne:

```html
@xalert("","","mt-4")
```

It is not an elegant solution but it works.

However, on top of it, I added named arguments.  So, it is possible to create a new component using named arguments.

```html
@xalert(type="error" message=$message class="mt-4")
```

so we could only specify the class as:

```html
@xalert(class="mt-4")
```

It is optional and it depends on how the tag is implemented.

But to the point:

## **What is the roadmap of this library?**

* Since I am using this library for my projects for my customers, then this library will not be discontinued. If not, then it always will be available.

* But neither Laravel's Blade nor BladeOne requires or needs new functionalities, at least in the core. Laravel's Blade hasn't added new functionalities (excluding components and functionalities specific of Laravel) for a long while.

* **So, the current version is the LTS version.**  Laravel's Blade is tied to a framework so they usually break the compatibility (to keep the compatibility with the newest changes). 

* BladeOne is different since it doesn't have any dependency, so it doesn't need to break compatibilities to keep the code running with other updated components. It even supports PHP 5.6 without affecting the performance.   

 It is the reason why BladeOne is still in version 3  (3.44) while Laravel's Blade is in version 7 (7.11). BladeOne supports practically all the functionalities of Blade (and some of them are fastest and I am not kidding), the exceptions are libraries specific of Laravel and "X-Components"

* In short, the roadmap has more functionalities over the older ones while it keeps the compatibility with previous projects.
